docs/mobile-apps/app-storage.md

```diff
@@ -120,6 +120,12 @@ Each session is a "fresh" installation of your app, meaning, you will not be abl
 
 App storage recognizes \*.apk and \*.aab files as Android apps and \*.ipa or \*.zip files as iOS apps. \*.zip files (for simulator tests only) are parsed to determine whether a valid *.app bundle exists.
 
+:::caution AAB Files are Resigned in Espresso Tests
+With Espresso, if you supply an .aab file, Sauce Labs must extract the .apk during installation, which necessitates resigning and instrumenting both the `app` and `testApp` in order to maintain matching signatures, without which tests will fail.
```

Comment:
@sweeneyskirt-sl is it resigning AND instrumenting that it will do?
I thought resigning was what was needed but instrumenting (which is changing the byte code) would not be needed
cc @linafarid

Reply:
Originally I thought so, too, @enriquegh, but when I read the Jira description a little more carefully, it seems that the act of resigning automatically kicks off instrumentation, which is actually the cause of the original error. Instrumentation fails because the app signatures don't match. At least, that is my understanding of the ticket. Please let me know if I have misinterpreted -- happy to rewrite!

Reply:
Instrumentation is not needed for the `.aab` case. The important part is "Sauce Labs must extract the .apk during installation", and the `.apk` mentioned here needs to be signed before we can install it on the device. We can still instrument that generated `.apk`, but it's not necessary (we just check the app storage settings and do it if it was requested).

Reply:
This is from the Jira ticket:
```The solution to fix this issue will introduce a limitation to the feature. Since, we HAVE to resign the main APK and also the test APK with Sauce’s signature, customers will not be able to actively disable “instrumentation”. This will be the default for Espresso tests.```

@arunkumarsaucelabs -- could you please provide some clarification regarding exactly what is at play in the solution?
- Is the behavior only for Espresso, or does it apply to Appium android tests as well?
- Is instrumentation affected?
Thanks!

Reply:
@sweeneyskirt-sl generally any .aab file that we receive from customers will have its generated .apk signed by Sauce. Regardless if it is used for Live, Appium or Espresso tests. The problem is that we don't have a customer certificate and we can only install .apks if they are signed. So we use ours to sign the apks.

In the case where a customer still wants to use instrumentation (e.g. use image injection) our logic will make sure that this is respected. 
In the case where a customer does not want to use instrumentation at all we will still sign it with our signature, otherwise as you stated the test won't run.

So summary is, regardless if instrumentation is turned on or off for the .aab file - we will always sign both .apks with our signature.

I would remove the word "instrumentation" and "resigning" completely from here in order not to create confusion and use "sign" instead.


Hope I did not add more to the confusion...

Reply:
Thanks for the clarification, @linafarid , but I am still slightly unclear about this part: 
> "In the case where a customer does not want to use instrumentation at all we will still sign it with our signature, otherwise as you stated the test won't run."
Do you mean that we will sign the app without instrumenting it? If so, then I don't really understand how this is a "limitation" as described in the Jira ticket. What am I missing here? I apologize for being so dense on this -- I really want to understand!

Reply:
@sweeneyskirt-sl I think its a limitation in that if the signing of the app with our own signature breaks the app (this happens if the app has any sort of security SDKS meant to protect the app's integrity) then its a "tough luck" situation since we **need** to sign it with our own signature due to the way AAB works.

Reply:
yes, that is very correct. Thanks @enriquegh